src/psyclone/psyir/nodes/array_of_structures_member.py

```diff
@@ -87,25 +85,15 @@ def create(member_name, indices, inner_member, parent=None):
         :rtype: :py:class:`psyclone.psyir.nodes.ArrayOfStructuresMember`
 
         '''
-        return ArrayOfStructuresMember._create_array_member(
-            member_name, indices, inner_member, parent=parent)
-
-    @staticmethod
-    def _validate_child(position, child):
-        '''
-        :param int position: the position to be validated.
-        :param child: a child to be validated.
-        :type child: sub-class of :py:class:`psyclone.psyir.nodes.Node`
-
-        :return: whether the given child and position are valid for this node.
-        :rtype: bool
-
-        '''
-        if position == 0:
-            # The first child must be a Member
-            return isinstance(child, Member)
-        # All subsequent child must be array-index expressions
-        return isinstance(child, (DataNode, Range))
+        obj = ArrayOfStructuresMember(member_name, parent=parent)
+        # Add any child Member as the first child
```

Comment:
Maybe "Add the provided inner_member node as the first child"?

Reply:
Yes, that's better.